libs/api/repository/src/lib/gn4/selection/selection.service.ts

```diff
@@ -33,48 +33,43 @@ export class SelectionService {
     this.selectedRecordsIdentifiers$.next(filtered)
   }
 
-  selectRecords(records: CatalogRecord[]): Observable<void> {
-    const newIds = []
-    records.map((record) => {
-      newIds.push(record.uniqueIdentifier)
-    })
-    const apiResponse = this.selectionsApi.add(BUCKET_ID, newIds)
-    return apiResponse.pipe(
-      tap(() => {
-        this.addIdsToSelected(newIds)
-      }),
-      map(() => undefined)
-    )
+  selectRecords(records: CatalogRecord[]) {
+    const newIds = records.map((record) => record.uniqueIdentifier)
+    this.selectionsApi
+      .add(BUCKET_ID, newIds)
+      .pipe(
+        tap(() => {
+          this.addIdsToSelected(newIds)
+        })
```

Comment:
I'm not a huge fan of tap, and as I'm only discovering the code, I wonder:
- if addIdsToSelected is storing selected ids locally, why not read from the state instead
- if it's a facade above the state, why not use an effect instead?

Reply:
Right, I simplified the selection service to hopefully make it more readable in https://github.com/geonetwork/geonetwork-ui/pull/797/commits/b0eeb67aa92e2aece5f63009949bc0b6041cf2ae

This service provides a local state as well as handles interaction with the backend API, there is no NgRx store for this part because repository interfaces can have various implementations.